accepted/2020/workloads/workload-resolvers.md

```diff
@@ -0,0 +1,237 @@
+
+### Workload Manifest Targets Files
+
+In addition to the manifest file itself (likely in json format), the workload manifest package will include a `WorkloadManifest.targets` file which should import the necessary .targets files from the workload pack if a project depends on the workload.  The `WorkloadManifest.targets` files from all workload manifests will be imported into all projects that use Microsoft.NET.Sdk.
+
+The `WorkloadManifest.targets` should use MSBuild Sdk imports to import .targets files from an workload pack.  This should be done conditionally based on properties that determine whether the workload should be used.  For example, the Android workload manifest targets file might look like this:
+
+```xml
+<Import Project="Sdk.targets" Sdk="Xamarin.Android.Sdk"
+        Condition="'$(TargetPlatformIdentifier)' == 'android'" />
+```
+
+This indicates that the `Sdk.targets` file should be imported from the `Xamarin.Android.Sdk` workload pack if the `TargetPlatformIdentifier` is `android`.  The version of the `Xamarin.Android.Sdk` workload pack to use will be the one defined in the workload manifest json file.
+
+Because the workload manifest targets files will be imported by all projects, it is important to try to guard against bugs in these targets from causing issues with projects that don't intend to use the workload at all.  One way of reducing risk can be to split the logic into multiple targets files, where the targets files with more complex logic are only imported from the entry point workload manifest targets file based on a simple condition.  For example:
+
+```xml
+<!-- WorkloadManifest.targets (entry point): -->
+
+<Project>
+    <Import Project="Xamarin.Manifest.Android.Framework.targets" Condition="'$(TargetPlatformIdentifier)'=='android'" />
+    <Import Project="Xamarin.Manifest.iOS.Framework.targets" Condition="'$(TargetPlatformIdentifier)'=='ios'" />
+</Project>
+
+<!-- Xamarin.Manifest.Android.Framework.targets -->
+
+<Project>
+    <Import Project="Sdk.targets" Sdk="Xamarin.Android.Sdk" />
+    <Import Project="Sdk.targets" Sdk="Xamarin.Android.Aot.Sdk" Condition="'$(UseAot)'=='true'" />
+    <!--
+    Import any other sdks based on project state.
+    -->
+</Project>
+```
+
+In this example, the iOS and Android workloads both share the same workload manifest.  In addition to `WorkloadManifest.targets`, the manifest package includes separate targets files specific to iOS and Android.  The more complex logic for which workload packs to import from is guarded behind a simple `TargetPlatformIdentifier` check in the `WorkloadManifest.targets`.
```

Comment:
> In this example, the iOS and Android workloads both share the same workload manifest.

I assume we would not recommend this. Is it even a good idea to show this as an example?

Reply:
If iOS and Android share components, then they may have to share the same workload manifest (because you can't have one workload depend on a different version of a workload pack than another workload for the same band).

@mhutch, what are your thoughts?

Reply:
Good point. My concern is that Android and iOS are expected to have updates on different schedules. How do we achieve the following goals:

* lower costs for us -- avoid re-testing iOS for Android updates, and vice versa
* avoid UX challenges for customers -- iOS customers do not get update notifications for Android updates, and vice versa

These might both be .NET 6.0 challenges given our scoping for .NET 6.0, but they are descriptive of my primary concerns.

Reply:
I would not expect iOS and Android to have the same manifest. But iOS, macOS, tvOS, and watchOS might as they are built out of the same codebase.